docs/source/models.rst

```diff
@@ -111,6 +114,7 @@ ShuffleNet V2                     30.64           11.68
 MobileNet V2                      28.12           9.71
 ResNeXt-50-32x4d                  22.38           6.30
 ResNeXt-101-32x8d                 20.69           5.47
+MNASNet 1.0                       26.49           8.456
```

Comment:
Quick question:

I've tried testing the models with the `references/train.py` file, and got
```
 * Acc@1 73.456 Acc@5 91.510
```
which corresponds to `26.54` and `8.49`.

I wonder if there is a difference in our data, or if the model that I downloaded is not the same?

Reply:
I'll verify and report back (and train a better model if needed). Could be something as dumb as the wrong version of Pillow. To aid with investigation, please answer the following questions:
 - Are you using Pillow-SIMD? 
 - Are you compiling it from source? 
 - Are you using libjpeg-turbo8?

Reply:
And also, did you change the number of epochs when training? I think that particular model actually used more epochs.

Reply:
FWIW, I trained this 'b1' variant of mnasnet to 74.66 and the 'a1' SE variant to 75.45. They both took over 400 epochs, both using RMSprop with Google like h-params. Using EMA of the model weights was necessary to match/surpass the papers... 

Models only: https://github.com/rwightman/gen-efficientnet-pytorch#pretrained

Training code w/ models: https://github.com/rwightman/pytorch-image-models

Reply:
@depthwise I don't think that Pillow / the different image processing is the cause.
I've done extensive experiments in the past with multiple models, and they all were fairly insensitive to using PIL / OpenCV. If that is indeed the case, then I'd be surprised, and that could indicate that the model is very fragile to small perturbations.

For completeness, I used Pillow 5.4.1, from pip, using libjpeg-turbo8

@rwightman interesting, and what happens if you do not apply EMA, do you recall what accuracies you get?

Reply:
@fmassa I was topping out around where Dmitry ended up, in the 73-73.9 range over 4 attempts. After adding my EMA impl, I resumed my best training sessions from a little before where the validation error and started to level off, and ran it with EMA for another 60-80 epochs ( think I went back to 380 and ended at 450 but have to check) and ending up getting the results I was looking for. I brought the dropout and weight decay back up to paper level when doing this, I had been lowering them before to squeeze out a few .1%  but the EMA smoothed weights benefit from having a little more variability. I keep the color jitter low, in the google training it looks like it's completely disabled for these models.

Reply:
So to investigate this I wrote a simple eval script, which I pushed to <https://github.com/1e100/mnasnet_trainer/blob/master/eval.py>.

The results with Pillow-SIMD/libjpeg-turbo-8 are as follows:
```
Dev/mnasnet_trainer % ./eval.py
Evaluating pretrained mnasnet1_0
1.0769143749256522 [('prec1', 73.490265), ('prec5', 91.53294)]
Evaluating pretrained mnasnet0_5
1.3720299355229553 [('prec1', 67.59815), ('prec5', 87.51842)]
```
Neither of which matches the published numbers exactly. MNASNet 1.0 is slightly worse than in the doc says. MNASNet 0.5 is slightly better than the checkpoint name would imply (67.598% top1 vs 67.592).

The results with "plain" pillow 6.0.0 from PyPl are as follows:
```
% ./eval.py
Evaluating pretrained mnasnet1_0
1.0772113243536072 [('prec1', 73.46037), ('prec5', 91.52099)]
Evaluating pretrained mnasnet0_5
1.372453243756781 [('prec1', 67.606026), ('prec5', 87.50845)]
```
So the top1 for 1.0 gets a bit worse, and for 0.5 it gets a bit better. I've observed such sensitivity with other "efficient" models in the past. In particular, the resize algorithm (which is different in Pillow SIMD) seems to make a noticeable difference. The effect of this on smaller models is easily measurable. Something as mundane as a different JPEG decoder affects them, and so do the software versions: CUDA/cuDNN, PyTorch, etc - a number of these are different between then and now.

Just to be on the safe side, though, I kicked off another run for 1.0. I should have the results sometime over the weekend, one way or the other.

Reply:
@1e100 if you are running the training again, would you mind using the code from `references/classification`, and maybe only changing the lr step?

For MobileNet V2, I used the following commands, on 8 GPUs
```
--model mobilenet_v2 --epochs 300 --lr 0.045 --wd 0.00004 --lr-step-size 1 --lr-gamma 0.98
```

The most important thing we are trying to do here is to have a simple path for reproducible research, so having a few % worse accuracy (~0.3-0.5 e.g.,) but with a reproducible script available in torchvision would be preferable I'd say.

This way, we can indeed compare apples with apples

Reply:
@fmassa what version of Pillow should I use for this? I use custom-compiled PillowSIMD for my own training runs, but if we want things to be more repeatable, I could use the slower, default install of PillowSIMD.

Here's how I compile it on my machines:
```bash
CC="cc -march=native -mtune=native -O3" pip3 install \
    --force-reinstall --user pillow-simd
```

Reply:
@1e100 sorry, I missed your message.

You can use current version that you have available, don't worry about potentially small differences.

Reply:
Still training. First training run wasn't too good (I tweaked the training regime, and this model is sensitive to that kind of thing), so I'm using that as initialization on the second run. I'll update (and send a PR) when I get a good result.